pkg/resourcequotas/resourcequotas.go

```diff
@@ -0,0 +1,198 @@
+package resourcequotas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1Typed "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// Builder provides struct for resource quotas containing connection to the cluster.
+type Builder struct {
+	// Resource Quota definition
+	Definition *corev1.ResourceQuota
+
+	// Created resource quota object
+	Object *corev1.ResourceQuota
+
+	// Used in functions that define or mutate deployment definition. errorMsg is processed before the deployment
+	// object is created.
+	errorMsg  string
+	apiClient corev1Typed.CoreV1Interface
+}
+
+// NewBuilder creates a new resource quota builder.
+func NewBuilder(apiClient corev1Typed.CoreV1Interface, name, nsname string,
```

Comment:
is there a reason the entire definition needs to be provided when creating a new builder? It also means that the name and nsname parameters do nothing

Reply:
You're right.  I was working with my existing code in my QE repo and there we define it completely before creating.  So I think what I'll do is create a base `ResourceQuota` object and do a `WithXYZ` func to add additional spec options.

Reply:
Done.